src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java

```diff
@@ -166,4 +167,42 @@ public Profile getLoggedInUser(long id)
         }
         return mentees;
     }
+
+    /**
+     * Update {@link EnrolmentState} of Approved {@link Mentee} objects of a user as Rejected
+     * to keep only one approved {@link Mentee}
+     *
+     * @param mentorId  which is the {@link Mentor} object's id of the {@link Mentee}
+     * @param profileId which is the id of the {@link Profile}
+     * @return the updated {@link Mentee}
+     *
+     * @throws ResourceNotFoundException is thrown if the {@link Mentor} doesn't exist
+     * @throws BadRequestException       is thrown if the {@link Mentee} is not approved
+     */
+    public Mentee confirmMentor(long mentorId, long profileId)
+            throws ResourceNotFoundException, BadRequestException {
+        Optional<Mentor> optionalMentor = mentorRepository.findById(mentorId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Mentor by id: " + mentorId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        Optional<Mentee> optionalMentee = menteeRepository.findByProfileIdAndMentorId(profileId, mentorId);
+        if (!optionalMentee.isPresent()) {
+            String msg = "Error, User hasn't applied for mentor by id: " + mentorId + ".";
+            log.error(msg);
+            throw new BadRequestException(msg);
+        }
+
+        if (!optionalMentee.get().getState().equals(EnrolmentState.APPROVED)) {
+            String msg = "Error, User is not approved by the mentor by id: " + mentorId + ".";
```

Comment:
```suggestion
            String msg = "Error, Mentee with id: " + menteeId + " is not approved by the mentor with id: " + mentorId + ".";
```

Reply:
We don't have the menteeId here. How about this?
```suggestion
            String msg = "Error, User with id " + profileId + " is not approved by the mentor by id: " + mentorId + ".";
```

Reply:
sure 👍

Reply:
It's already updated